### ISSUE TITLE
Make `marimo` more like `Jupyter Lab`

### ISSUE THREAD

**Issue by toiletsandpaper** [marimo-team/marimo#1980]
### Description

When me and my team developing ML-based solutions - we use Jupyter Lab or IDEs, that acts like Jupyter Lab (VSCode + extensions or DataSpell), because it much easier to edit and inspect several files at once. For example - we use project structure like this https://github.com/drivendataorg/cookiecutter-data-science 

But marimo is acts more like Jupyter Notebook. Notebook is old and uncomfortable way to develop. Lab is already act like a IDE.

### Suggested solution

So, I'm asking marimo team to change the current vector from `Notebook` to `Lab`. Or maybe combine somehow in your own way?

But for now, marimo is great for presenting experiment and EDA, but not ready for production.

### Alternative

- Jupyter Lab
- VSCode + Jupyter Extensions
- IntelliJ DataSpell

**Comment by dmadisetti** [comment]
Thanks for the feedback.

Do you use marimo's home feature? i.e. running marimo edit on a directory which lets you navigate between notebooks? Otherwise, the marimo vs-code plugin can give you a direct IDE experience.

Otherwise, could you be more specific about lab features that are missing from marimo?

I know some of my frustration specific to marimo over Jupyter lab are:
- not being able to easily navigate directly from one notebook to another
- not being able to navigate back to home easily
- not being able to have more than 1 tab open of the same notebook

marimo does have a file editor builtin, it's not ide-level but it's comparable to colab's editor. Is this something you use in Lab a lot?

**Comment by toiletsandpaper** [comment]
> Do you use marimo's home feature?

Yep, but its kinda strange to see nothing when you start new project in `Workspace` panel. I want to see all my folders, check other files and etc. Maybe I want to create a `001_preprocess.marimo.py`, `002_eda.marimo.py`, `003_train.marimo.py` files and only after this open 001 and edit? And even after I create - home page only shows marimo notebooks, not file tree view to explore and edit.

> marimo vs-code plugin can give you a direct IDE experience.

There are two main reasons why we dont work like that:
1. We work on remote machines in closed from internet access environments, so VSCode SSH Remote Development extensions + Marimo extensions is super unstsable or even uninstallable at all.
2. Marimo extensions for VSCode have really strange design with `side-by-side` view. Maybe `only marimo view` option will fix that.

> marimo does have a file editor builtin, it's not ide-level but it's comparable to colab's editor. Is this something you use in Lab a lot?

Of course! For example, editing `.env` files for S3 connections or `.yaml` and `.toml` for configuring other tools! Auto-completions is not that important for such files, just open it in full-size editor.

<hr>

But yeah, all the problems comes from working in cloud and using marimo as IDE, not just `1-tab editor`

**Comment by majidaldo** [comment]
i say keep the nb experience separate from the ide experience.

**Comment by toiletsandpaper** [comment]
> i say keep the nb experience separate from the ide experience.

IMO, nb experience is only about working in one notebook, without additional tools. Marimo can be much more like Lab, keeping an excellent Notebook experience.
